### PR TITLE
Script to test websocket.go

### PR DESCRIPTION
**Pull request by maurya-anuj**
It contains Unit Test file for Websocket.go file

**What type of PR is this?**
 test

**What this PR does / why we need it**:
Adds unit Test file for Websocket.go file in pkg/edgehub/clients/wsclient

### CODE REVIEW

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+//Message is content object to be passed in as message object
+type Message struct {
+	Name string `json:"name"`
+}
+
+var testServer *httptest.Server
+var upgrader = websocket.Upgrader{}
+
+//newTestServer() starts a fake server for testing
+func newTestServer() {
```
Comment (rohitsardesai83) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: it's better if we move this method to a websocket helper file . if any other tests need to start a fake web socket server. this method will be useful . @pavan187 , please guide.

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+import (
+	"net/http"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
```
Comment (sids-b) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: seperate kubeedge, vendor and golang packages

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+				err := c.ReadJSON(&m)
+				if err != nil {
+					break
+				}
+				err = c.WriteMessage(3, []byte(""))
+				if err != nil {
+					break
+				}
+			}
+
```
Comment (sids-b) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: remove blank line

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+	}
+}
+
+//Message is content object to be passed in as message object
+type Message struct {
+	Name string `json:"name"`
+}
+
+var testServer *httptest.Server
+var upgrader = websocket.Upgrader{}
```
Comment (sids-b) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: = can be removed ?

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+					ExtendHeader:     http.Header{},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			wcc := &WebSocketClient{
+				webConn:  tt.fields.webConn,
+				sendLock: tt.fields.sendLock,
```
Comment (sids-b) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: no need to declare new structure here, can directly use tt.fields to call the function ??

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+				config:   tt.fields.config,
+			}
+			wcc.Init()
+			wcc.Uninit()
+			err := wcc.webConn.WriteMessage(2, []byte(""))
+			if err == nil {
+				t.Errorf("WebSocketClient.Uninit")
```
Comment (sids-b) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: can we have a negative testcase as well ??

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			wcc := &WebSocketClient{
+				webConn:  tt.fields.webConn,
+				sendLock: tt.fields.sendLock,
+				config:   tt.fields.config,
+			}
+			if err := wcc.Init(); (err != nil) != tt.wantErr {
```
Comment (sids-b) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: can we assert the error here ?

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+			wcc := &WebSocketClient{
+				webConn:  tt.fields.webConn,
+				sendLock: tt.fields.sendLock,
+				config:   tt.fields.config,
+			}
+
+			//First run init
+			wcc.Init()
+
+			if err := wcc.Send(tt.message); (err != nil) != tt.wantErr {
```
Comment (sids-b) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: Can we assert the error here ??

File: pkg/edgehub/clients/wsclient/websocket_test.go (hunk 1 above)
Comment (pavan187) [reply pkg/edgehub/clients/wsclient/websocket_test.go]: @rohitsardesai83  this function starting the websocker server on some hard coded api's, which can be used only to this test file, there are no standard/common api's this method is using, i feel other test functions cannnot reuse this method for websocker server testing.

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/satori/go.uuid"
+
+	"fmt"
```
Comment (sids-b) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: nit: move fmt in the top batch.

File: pkg/edgehub/clients/wsclient/websocket_test.go (hunk 1 above)
Comment (rohitsardesai83) [reply pkg/edgehub/clients/wsclient/websocket_test.go]: ok.

File: pkg/edgehub/clients/wsclient/websocket_test.go [hunk 10]
```diff
@@ -0,0 +1,352 @@
...
+	tests := []struct {
+		name          string
+		fields        WebSocketClient
+		expectedError error
+	}{
+		{name: "TestInit: Success in connection ",
+			fields: WebSocketClient{
+				webConn:  &websocket.Conn{},
+				sendLock: sync.Mutex{},
+				config: &WebSocketConfig{
```
Comment (rohitsardesai83) [inline pkg/edgehub/clients/wsclient/websocket_test.go]: the creation of the websocketconfig object can be a separate method with parameters  , as all tests create this.

**Pull request merged**